### PR TITLE
Add Musca-A & B port to DAPLink

### PR DESCRIPTION
**Pull request by avinashmehtadelhi**

### CODE REVIEW

File: test/info.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #
 # DAPLink Interface Firmware
-# Copyright (c) 2009-2016, ARM Limited, All Rights Reserved
+# Copyright (c) 2009-2019, ARM Limited, All Rights Reserved
```
Comment (0xc0170) [inline test/info.py]: Please revert changing permissions to this file (100644 → 100755)

Comment (0xc0170) [reply test/info.py]: Few files in this PR

File: source/family/arm/musca_a/target_reset.c [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    buf[6] = 0x00;
+    buf[7] = 0x10;
+    buf[8] = 0xFE;
+    buf[9] = 0xE7;
+    buf[10] = 0xFE;
+    buf[11] = 0xE7;
+
+    swd_write_memory(0x10000000, (uint8_t *)buf, 12);
+
+//    swd_write_word(0x10000000, 0x30008000);
```
Comment (0xc0170) [inline source/family/arm/musca_a/target_reset.c]: why is this dead code left here?

File: source/daplink/interface/main.c [hunk 3]
```diff
@@ -363,6 +366,29 @@ void main_task(void * arg)
                 reset_pressed = 0;
             }
 
+#ifdef PBON_BUTTON
+            // handle PBON pressed
```
Comment (0xc0170) [inline source/daplink/interface/main.c]: I've seen some comments about `PBON_BUTTON`  - why is this needed here and how this can be used generically ?

File: test/info.py (hunk 1 above)
Comment (avinashmehtadelhi) [reply test/info.py]: changed

File: source/family/arm/musca_a/target_reset.c (hunk 2 above)
Comment (avinashmehtadelhi) [reply source/family/arm/musca_a/target_reset.c]: removed now.

File: source/daplink/interface/main.c (hunk 3 above)
Comment (avinashmehtadelhi) [reply source/daplink/interface/main.c]: targets with Power button (like musca) can #define PBON_BUTTON in there override file and power button handling will be compiled for those boards.

**Comment by 0xc0170** [comment]
@brianesquilona @flit looks fine to me, not certain about the button addition - can you please review?

**Comment by 0xc0170** [comment]
Let's progress here.

I still dont like the change for PBON_BUTTON - will get this resolved on master somehow.

File: source/daplink/interface/main.c (hunk 3 above)
Comment (0xc0170) [reply source/daplink/interface/main.c]: I read the comments in previous PR -is this really needed ? If I remove this ifdef from here - what is a result? Why there is this button check, it's not common funcitonality.

Even with ifdef, this is polluting common code. I would rather have a hook - functionality there to hook into than ifdef. I merged this to progress , but still see this as unresolved
